### PR TITLE
Fixing bug where all users’ AILTN logs are being uploaded

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m [hunk 1]
```diff
@@ -288,6 +288,11 @@ - (BOOL) readAnalyticsPolicy {
 }
 
 - (void) publishOnAppBackground {
+
+    // Publishing should only happen for the current user, not for all users signed in.
+    if (![self.userAccount.idData isEqual:[SFUserAccountManager sharedInstance].currentUser.idData]) {
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m]: Shouldn't this comparison be on `accountIdentity`?

Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m]: Argh. Forgot to commit. Yes.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m [hunk 2]
```diff
@@ -288,6 +288,11 @@ - (BOOL) readAnalyticsPolicy {
 }
 
 - (void) publishOnAppBackground {
+
+    // Publishing should only happen for the current user, not for all users signed in.
+    if (![self.userAccount.accountIdentity isEqual:[SFUserAccountManager sharedInstance].currentUser.accountIdentity]) {
+        return;
+    }
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m]: What if the dispatch_async is called and the user switch, would the events still be published under the wrong user then?

**Comment by khawkins** [comment]
The only other thing that I'd say is that this is an artificial limitation that should be addressed more comprehensively at some point.  To wit, we've siloed some of the analytics management by user, while de facto relying on `currentUser` in other parts of the system (notably, here: publishing).

Ideally, it should all be siloed by user, so that current user context doesn't matter for things like publishing.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m (hunk 2 above)
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m]: This call gets triggered after app is backgrounded - user switch can't happen then.

**Comment by bhariharan** [comment]
@khawkins I agree - there's a larger body of work I plan on doing in the `5.2` timeframe on this.

**Comment by huminzhi** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Code coverage under minimum of 80%</td>
    </tr>
  </tbody>
</table>

## Current coverage for SalesforceAnalytics.framework is `66%`
✅ *No files affecting coverage found*

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**